### PR TITLE
Reader: Push a page view when someone likes a post in the stream

### PR DESCRIPTION
**Pull request by blowery**
Solves a long-standing question from folks who ask 'why do I have more likes than page views?'. Consider a like to be enough to push a page view.

cc @nbradbury @aerych

### CODE REVIEW

**Comment by designsimply** [comment]
Tested `add/page-views-for-likes-from-streams ` with Firefox 45.0.1 on Mac OS X 10.11.13

Video: [5m24s](https://cloudup.com/cK-breXiZuE) (might have to turn it up, audio is ick)

My testing steps:

1. Publish some brand new posts on a test blog (note: helps to use a test blog with little or no traffic so you can spot the stats changes easily)
1. Log out and log back in as a different user
1. Follow the blog from step 1
1. Go to http://calypso.localhost:3000 with the `add/page-views-for-likes-from-streams ` branch active
1. Like the posts published in step 1
1. View a few of the posts from step 1 in full post view
1. Log back in as the blog owner from step one and review the stats on the Insights, Days, Weeks, Months, and Years summary pages

Problems found:

* Likes and views were missing from the Days summary but were present in the Weeks/Months/Years summaries
* Insights "Latest Post Summary" showed 0 views because views were registered for the home page instead of for each post—should the views go to each post since the accompanying likes were specific to each post?
* Full post views were not counted at all (possible separate bug)

**Comment by blowery** [comment]
@designsimply Nice catch on the URL we're pushing page views to! I'll get that fixed.

The reason you're not seeing page views for the subsequent full post views at the moment is because I'm limiting the page view to one per session per post. I'm working out a way to get around that, it was just the simplest way forward.

**Comment by blowery** [comment]
> The reason you're not seeing page views for the subsequent full post views at the moment is because I'm limiting the page view to one per session per post. I'm working out a way to get around that, it was just the simplest way forward.

Actually, I'm wrong. It should work as is. In fact, it should push a page view every time you like a post (so toggling like on a post would push a page view every time you turned it "on").

**Comment by blowery** [comment]
> Nice catch on the URL we're pushing page views to! I'll get that fixed.

fixed in https://github.com/Automattic/wp-calypso/pull/4482/commits/b72aeceff473a15ae83e8a34300a2eb0e7a6408f

**Comment by blowery** [comment]
Ah, I see.

So, we don't actually push page views ever from calypso.localhost, only from production. The way to verify this ticket is:

1. Spin up calypso
2. Pull up the dev console
3. turn on debug traces for reader stats `localStorage.debug = 'calypso:reader:stats';`
4. reload the page
5. take the various actions that should push page views and verify `calypso:reader:stats reader page view for post` logs appear where appropriate

**Comment by blowery** [comment]
> fixed in b72aece

I unfixed this, squashed commits, and force pushed. We already repost the post ID when we push the page view event. The real issue here was that we don't report page views from development environments.

**Comment by designsimply** [comment]
Tested again with `add/page-views-for-likes-from-streams ` using Firefox 45.0.1 on Mac OS X 10.11.13 and I feel like either it's not working or the troubles you mentioned about stats + dev env are my nemesis right now… 

My steps:

1. Stop Calypso
1. `git checkout master`
1. `git pull`
1. `git branch -D add/page-views-for-likes-from-streams `
1. `git checkout add/page-views-for-likes-from-streams `
1. `make run`
1. Log in as the blog owner and check stats so we can compare numbers later
1. Log out and log back in as another user
1. Run this command in the console: `localStorage.debug = 'calypso:reader:stats';` and refresh
1. Follow the blog from step 7 and like the latest post for that blog in the main Reader stream
1. Log out and log back in as the blog owner
1. Check stats to see if a view was registered in Insights/Days/Weeks stats

Results: a like was counted, but not a view—console logs looked okay though, maybe that's all that needed checking from local?

Video: [2m44s](https://cloudup.com/c006pew6YkO)

Stats before:

screen shot 2016-04-04 at mon apr 4 3 31 40 pm

Console output:

screen shot 2016-04-04 at mon apr 4 3 32 37 pm

Stats after:

screen shot 2016-04-04 at mon apr 4 3 31 57 pm

**Comment by blowery** [comment]
> Results: a like was counted, but not a view—console logs looked okay though, maybe that's all that needed checking from local?

Yup, that's the expected result. There's no way to really see the results of pushing stats from dev, we only test that the stat was going to be sent correctly.

**Comment by bluefuton** [comment]
You also need to be a non-admin user to see this in action (`markPostSeen` does not fire stats if you're an admin). 

Works well for me. Fires the correct event on like:

screen shot 2016-04-05 at 15 31 10

File: client/lib/feed-stream-store/test/analytics/index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
```
Comment (bluefuton) [inline client/lib/feed-stream-store/test/analytics/index.js]: Do we want to add a test here? :)

File: client/reader/following-stream/post.jsx [hunk 2]
```diff
@@ -335,6 +336,12 @@ const Post = React.createClass( {
 		}
 	},
 
+	maybeMarkPostSeen: function( liked ) {
+		if ( liked ) {
+			FeedPostStoreActions.markSeen( this.props.post, 'stream' );
```
Comment (bluefuton) [inline client/reader/following-stream/post.jsx]: We don't seem to be using the second param - can we drop it?

File: client/reader/full-post/index.jsx [hunk 3]
```diff
@@ -444,15 +427,12 @@ FullPostContainer = React.createClass( {
...
-				site && site.get( 'state' ) === SiteState.COMPLETE ) {
-			if ( site.get( 'is_private' ) || isNotAdmin ) {
-				readerPageView( site.get( 'ID' ), site.get( 'URL' ), post.ID, site.get( 'is_private' ) );
-			}
+			site = this.state.site;
+
+		if ( post && post._state !== 'pending' &&
+			site && site.get( 'state' ) === SiteState.COMPLETE &&
+			! this.hasSentPageView ) {
+			PostStoreActions.markSeen( this.state.post, 'full-post' );
```
Comment (bluefuton) [inline client/reader/full-post/index.jsx]: We don't seem to be using the second param - can we drop it?

**Comment by bluefuton** [comment]
Couple of tiny nits but otherwise looks ready to :ship:

**Comment by bluefuton** [comment]
Couple of tiny nits but otherwise looks ready to :ship:

File: client/lib/feed-stream-store/test/analytics/index.js (hunk 1 above)
Comment (blowery) [inline client/lib/feed-stream-store/test/analytics/index.js]: nope, this is just stub for the real analytics module

File: client/reader/following-stream/post.jsx (hunk 2 above)
Comment (blowery) [inline client/reader/following-stream/post.jsx]: yup, will do

File: client/lib/feed-stream-store/test/analytics/index.js (hunk 1 above)
Comment (bluefuton) [inline client/lib/feed-stream-store/test/analytics/index.js]: Ah, it's a stub. I am a narna.

**Pull request merged**